YaLibs/YaToolsIDALib/Repository.cpp

```diff
@@ -750,20 +770,34 @@ bool Repository::rebase(const std::string& upstream, const std::string& destinat
 
 bool Repository::add_file_to_index(const std::string& path)
 {
+    fs::path p = path;
+    if (!include_idb_ &&
+        (path == get_current_idb_name() ||
+        path == get_original_idb_name()))
+    {
+        p = ".gitignore";
+        f.open(p, std::fstream::out);
+        f << get_current_idb_path() << std::endl;
+    }
     try
     {
-        repo_.add_file(path);
+        repo_.add_file(p.generic_string());
     }
     catch (const std::runtime_error& error)
     {
-        LOG(WARNING, "Failed to add %s to index, error: %s\n", path.c_str(), error.what());
+        LOG(WARNING, "Failed to add %s to index, error: %s\n", p.generic_string().c_str(), error.what());
         return false;
     }
     return true;
 }
 
 bool Repository::remove_file_for_index(const std::string& path)
 {
+    if (!include_idb_ &&
```

Comment:
why is it needed to change this function ?

Reply:
It isn't...fixed in 8000c16d257bb85046304531e5aac8ea1c3fe6f4